torchbenchmark/util/distributed/trainer.py

```diff
@@ -36,7 +36,7 @@ def __init__(self, args, model_class, mode="SPMD", model_args=None):
 
         # create model instance after Trainer setup, so that
         # visible devices won't be revised in model constructor
-        self.e2e_benchmark: E2EBenchmarkModel = model_class("train", known_args.device, batch_size=None, extra_args=extra_args)
+        self.e2e_benchmark: E2EBenchmarkModel = model_class("train", device=known_args.device, batch_size=None, extra_args=extra_args)
```

Comment:
I think you can just remove the `device` parameter. We leave the device choosing to the model code.